### PR TITLE
TTD Windows music fixups

### PR DESCRIPTION
**Pull request by nielsmh**
This patchset tries to work around two issues with the MIDI files included in TTD for Windows: The title song has a bit of delay at the beginning, causing it to loop poorly, and "Can't get there from here" has a very long silence at the end, which is not there in the original DOS version music.

### CODE REVIEW

File: src/music/dmusic.cpp [hunk 1]
```diff
@@ -832,8 +832,9 @@ static void MidiThreadProc(void *)
 			/* end? */
 			if (current_block == current_file.blocks.size()) {
 				if (current_segment.loop) {
-					current_block = 0;
+					current_block = current_segment.start_block;
 					clock->GetTime(&playback_start_time);
+					playback_start_time -= current_file.blocks[current_block].realtime * MIDITIME_TO_REFTIME;
```
Comment (nielsmh) [inline src/music/dmusic.cpp]: I haven't been able to figure out why yet, but the DirectMusic driver loops the title song _too_ tightly, producing a different wrong-sounding result.

Comment (michicc) [reply src/music/dmusic.cpp]: I more or less cargo-culted a lot from your MME driver; I might have messed stuff up.

Comment (michicc) [reply src/music/dmusic.cpp]: Hmm, actually, DMusic sequences all by messages by the absolute timestamp that is sent. If you subtract too much from the playback start time, the new messages might be sequenced before the old messages.

You might try setting the playback start time to `block_time`, i.e. the last transmitted timestamp and not the current time.

Comment (nielsmh) [reply src/music/dmusic.cpp]: Yep basing it off `block_time` fixes the issue.

**Review by michicc — APPROVED** [review]

**Pull request merged**